src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,51 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_LOGIN_LENGTH = 6;
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private static final int MIN_AGE = 18;
+
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
-    public User register(User user) {
-        return null;
+    public User register(User user) throws RegistrationException {
+        String login = user.getLogin();
+        if (login == null) {
+            throw new RegistrationException("Login cannot be null!");
+        }
+        if (login.isEmpty()) {
+            throw new RegistrationException("Login can't be empty");
+        }
+        if (login.length() < MIN_LOGIN_LENGTH) {
+            throw new RegistrationException("Login must be at least "
+                    + MIN_LOGIN_LENGTH + " characters long");
```

Comment:
It's good that you're checking the login for null and length. However, according to the checklist, we should avoid hardcoding values that may change over time. Consider if there's a need for a maximum login length or if the minimum length should be configurable in case the requirements change in the future.